frame.go

```diff
@@ -1717,25 +1759,46 @@ func (f *framer) writeBatchFrame(streamID int, w *writeBatchFrame, customPayload
 		if w.defaultTimestamp {
 			flags |= flagDefaultTimestamp
 		}
+	}
 
-		if f.proto > protoVersion4 {
-			f.writeUint(uint32(flags))
+	if f.proto > protoVersion4 {
+		if w.keyspace != "" {
+			flags |= flagWithKeyspace
```

Comment:
same as above, panic or return error if user tries to set keyspace but protocol is not > 4

Reply:
The same as above.